### DISCUSSION TITLE
Make it much easier for clients to upgrade/buy Craft

### DISCUSSION THREAD

**Discussion by darylknight** [craftcms/cms#8922] [Ideas]
I've had a client sign up for an account at https://id.craftcms.com and I've added the automatically generated license key to their account (which is Solo). Now that we're using Project Config, it's no longer possible to temporarily upgrade a client's CMS account to a full Admin to get them to buy a license through the control panel, which means the only option is the external plugin Store. There's no way to upgrade an existing key from the Solo key in id.craftcms.com.

As far as I can tell, the only thing they can do is Buy Craft CMS Pro from the bottom of the Plugin Store page, which then adds a second license key to their account and I'll then replace the exising Solo key on the site with the new one.

Just in general, could I please request that you make it way easier for clients to buy licenses for Craft and plugins? It feels very developer-centric at the moment, and we don't really want to be asking clients for their credit cards to do it ourselves.

**Comment by brandonkelly** [comment]
Yep, we are working on improving this.

**Comment by siebird** [comment]
I agree with @darylknight, it's currently a broken process with the project config mandate in 3.5. 

I have to temporarily allow admin changes on production for this to happen > give client instructions on purchasing > then commit changes on production > push back to origin and sync down.

**Comment by darylknight** [comment]
No, you shouldn't be making changes on production. Get them to send you the license keys after buying through the plugin store and add them to your local machine

**Comment by siebird** [comment]
Not sure I agree with that workflow. It needs to be automated. The more steps, the more broken/cumbersome the process will be.

**Comment by bymayo** [comment]
Yeah, only being able to add license keys to the site on local is not a great workflow. We've had this now with a few clients and everytime the process feels a bit messy.

Ideally it would be great if plugin licenses were separate from the settings, so admins can still add keys even when `allowAdminChanges` is `false`

Even being able to update/see the license key of Craft itself would be great as currently this is only possible to update via the license.key file and sometimes these keys don't match up with what you purchase.

**Comment by brandonkelly** [comment]
We improved this in 3.6, actually. Now commercial plugins are automatically assigned trial license keys when installed (or when you visit the **Settings** → **Plugins** page), and you are able to purchase Craft Pro and any commercial plugins that already have trial license keys from your production environment.

There’s still room for improvement (purchasing the licenses still requires an admin account), but it’s a lot better now than it was before.

**Comment by bymayo** [comment]
Thanks for the update. My issues still seem to be in 3.6 really. Especially when license keys are conflicting and need updating etc. E.g. At the moment I purchased licenses for Craft + plugins, but it's still showing I need to buy a Craft license and a license for one plugin?! Even though my invoice is showing it as purchased.

But, I can't see what license key is set for the plugin and update it as it's hidden in settings.

**Comment by brandonkelly** [comment]
What do you mean by “license keys are conflicting”? Did you purchase within the control panel or from [plugins.craftcms.com](https://plugins.craftcms.com)?

- If within the control panel, was it locally or on production? And if on production, do you have `allowAdminChanges` disabled there or not?

- If from [plugins.craftcms.com](https://plugins.craftcms.com), the order that you add the new license keys is important. You must save your new `config/license.key` file **first**, and then paste in your plugin license keys. If you paste in your plugin license keys first, those licenses will become assigned to the previous Craft (Solo) license, and you will need to log into [id.craftcms.com](https://id.craftcms.com) and release them from it.

**Comment by bymayo** [comment]
Purchased through the control panel on production with `allowAdminChanges` disabled. I noticed when the payment went through there was a 500 error appeared in console, so maybe this was a glitch in applying the license keys or getting something back from Craft ID as a successful payment, so might not be specifically a Craft issue, but Craft ID.

Either way, I think being able to see / set the licenses manually via the CP would be the best solution.

**Comment by brandonkelly** [comment]
Can you search your `storage/logs/` folder on the server for `[error]` and see if there’s anything in there around the same timestamp as when this happened?

**Comment by darylknight** [comment]
I've just emailed support about this again for a new project (thanks for replying so quickly!) but thought I'd bump this too as it looks like the original issue was added to a milestone to 4.0, then closed. Are there any plans to improve the process of clients being able to buy their own licenses?

The situation we're in now is that a client wants to upgrade their Freeform license from Lite to Pro, and their only option, even if I make them an administrator (which is not ideal) is to buy a new license for Pro, unless I disable Project Config. Buying licenses is one of the first things a client will do with Craft CMS, and it really doesn't feel like a good experience when you've just sold them on the idea of moving away from WordPress. Craft is awesome, but this has been a point of difficulty for a very long time.

**Reply by brandonkelly** [reply]
It’s gotten a lot better for Craft Pro + plugin licenses in 3.6, however plugin upgrades do need to be ironed out more.
